### DISCUSSION TITLE
How to configure the access level of a Mapper object imported by another Mapper

### DISCUSSION THREAD

**Discussion by ponyjoy** [mapstruct/mapstruct#3493] [Q&A] [answered]
I have an issue when using MapStruct. Let's say I have a FooMapper that references another mapper BarMapper like this:
``` java
@Mapper(componentModel = MappingConstants.ComponentModel.CDI, uses={BarMapper.class})

interface FooMapper{

DemoDto toDemoDto(Demo demo);

}

@Mapper(componentModel = MappingConstants.ComponentModel.CDI)
interface BarMapper{

//...

}
```

I'm using Quarkus as my DI framework, which uses CDI. In the generated FooMapperImpl class, the code looks like:
``` java
class FooMapperImpl implements FooMapper{

     // The generated code has private access
    @Inject
    private BarMapper barMapper;
    // ...
}
```
The injected barMapper field has private access. Quarkus recommends using default access for @Inject dependencies. This causes a compiler warning.

How can I resolve this and control the access level of the generated barMapper field?

**Comment by thunderhook** [comment, marked as answer]
Just to clarify the background of the warning from quarkus: https://quarkus.io/guides/cdi-reference#native-executables-and-private-members

As far as I know, there is currently no possibility to change the visibility of used mappers. I couldn't find anything in our documentation. You can raise an issue for this. We can then have a look at it. 👍🏻

**Comment by ponyjoy** [comment]
Thank you very much for your patient explanation. I just tried again, after setting the injectionStrategy of FooMapper to InjectionStrategy.CONSTRUCTOR, the generated code is as follows, which can solve this problem.

```java
class FooMapperImpl implements FooMapper{

	private BarMapper barMapper;

	public FooMapperImpl(BarMapper barMapper){
		this.barMapper = barMapper;
	}
	
}
```
Quarkus's complaints are gone.
